src/reducers/entities/teams.js

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {TeamsTypes, UsersTypes} from 'constants';
+
+function currentId(state = '', action) {
+    switch (action.type) {
+    case TeamsTypes.SELECT_TEAM:
+        return action.teamId;
+    default:
+        return state;
+    }
+}
+
+function teams(state = {}, action) {
+    switch (action.type) {
+    case TeamsTypes.RECEIVED_ALL_TEAMS:
+        return Object.assign({}, state, action.data);
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function myMembers(state = {}, action) {
+    const nextState = {...state};
+    switch (action.type) {
+    case TeamsTypes.RECEIVED_MY_TEAM_MEMBERS:
+        const members = action.data;
+        for (const m of members) {
+            nextState[m.team_id] = m;
+        }
+        return nextState;
+    case TeamsTypes.LEAVE_TEAM:
+        const data = action.data;
```

Comment:
I could do a pass for this myself since I'm being a bit picky, but I think we should consider not using `action.data` everywhere. I know that's the standard convention, but it's obscuring what data is passed by the action and making it hard to understand what each case is doing.